.github/workflows/jest_tests.yml

```diff
@@ -0,0 +1,41 @@
+name: Running all tests
+
+on:
+  pull_request:
+    branches: ['main',  'Develop', 'hotfixes', 'release']
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Testing WebApp
```

Comment:
What each of this steps does is cd to each directory and then run `npm install` and `npm run test`, if some test fails it wont let you merge.